enhancements/installer/aws-ingress-nlb-security-group.md

```diff
@@ -0,0 +1,501 @@
+	// and manage security groups for the Network Load Balancer.
+	//
+	// +optional
+	// +openshift:enable:FeatureGate=IngressNLBSecurityGroup
+	ManagedSecurityGroup bool `json:"managedSecurityGroup,omitempty"`
+}
+```
+
+- The CIO will set the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group: "true"` annotation on the default router Service if the `managedSecurityGroup` field in the IngressController spec is set to true.
+
+#### Installer
+
+- Introduce a new boolean field `managedSecurityGroup` under `platform.aws.ingressController` in the `install-config.yaml`. This field will only be considered when `platform.aws.lbType` is `NLB`.
+
+```go
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+	// IngressController is an optional extra configuration for the Ingress Controllers.
+	// +optional
+	IngressController *IngressController `json:"ingressController,omitempty"`
+}
+
+// IngressController specifies the additional ingress controller configuration.
+type IngressController struct {
+	// SecurityGroupEnabled is an optional field to enable security groups
+	// when using load balancer type Network Load Balancer (NLB).
+	// When this field is enabled with LBType NLB, all ingresscontrollers (including the
+	// default ingresscontroller) will be created using security group in the NLBs
+	// by default.
+	//
+	// If this field is not set explicitly, it defaults to no security groups in the NLB.
+	// This default is subject to change over time.
+	//
+	// +optional
+	SecurityGroupEnabled bool `json:"securityGroupEnabled,omitempty"`
+}
+```
+
+- TBD: deprecate the field `platform.aws.lbType` in favor of `platform.aws.ingressController.loadBalancerType`
+
+#### ROSA Classic
+
+- TBD: API changes in Hive to read and process the new install-config option.
+
+#### Hypershift/ROSA HCP
+
+- TBD: API changes in Hypershift to configure CIO based on the install-config option.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+- TODO/TBD: The flow needs to be explored to determine how the install-config option propagates to the Hosted Control Plane's CIO configuration. This might involve setting the annotation on the Service created in the guest cluster.
+
+#### Standalone Clusters
+
+<!-- Is the change relevant for standalone clusters? -->
+
+> TODO/TBD: All changes is proposed initially and exclusively for Standalone clusters.
+
+
+#### Single-node Deployments or MicroShift
+
+> TODO/TBD
+
+
+### Implementation Details/Notes/Constraints
+
+> WIP/TBReviewed
+
+- The implementation in CCM should handle the case where the `service.beta.kubernetes.io/aws-load-balancer-managed-security-group` annotation is set to `true` but the service type is not `NLB` (`aws-load-balancer-type: nlb`). In this scenario, the CCM should likely log a warning mentioning the annotation is supported only on NLB.
```

Comment:
What does the CCM do today for annotations that don't apply? I suspect it ignores them

We can use VAP downstream to prevent this